src/libs/ajax/Support.ts

```diff
@@ -0,0 +1,87 @@
+import _ from 'lodash/fp';
+import { fetchOk, jsonBody } from 'src/libs/ajax/ajax-common';
+
+export interface ZendeskCreateSupportRequestRequest {
+  name: string;
+  email: string;
+  currUrl: string;
+  subject: string;
+  type: 'question' | 'bug';
+  description: string;
+  attachmentToken: string;
+  emailAgreed: boolean;
+  clinicalUser: boolean;
+}
+
+export interface ZendeskCreateSupportRequestResponse {
+  request: {
+    custom_status_id: number;
+    description: string;
+    id: number;
+    status: string;
+    subject: string;
+  };
+}
+
+export interface ZendeskUploadResponse {
+  token: string;
+  expires_at: Date;
+  attachment: {
+    url: string;
+    id: number;
+    file_name: string;
+    content_url: string;
+    content_type: string;
+  };
+}
+
+export const Support = (signal?: AbortSignal) => {
+  return {
+    // If you are making changes to the Support Request Modal, make sure you test the following:
+    // 1. Submit a ticket via Terra while signed in and signed out
+    // 2. Check the tickets are generated on Zendesk
+    // 3. Reply internally (as a Light Agent) and make sure an email is not sent
+    // 4. Reply externally (ask one of the Comms team with Full Agent access) and make sure you receive an email
+    createSupportRequest: async (
+      request: ZendeskCreateSupportRequestRequest
+    ): Promise<ZendeskCreateSupportRequestResponse> => {
+      const { name, email, currUrl, subject, type, description, attachmentToken, emailAgreed, clinicalUser } = request;
+      return fetchOk(
```

Comment:
It looks like this returns a `Response` object. To match the type signature, it would need return the JSON decoded response text.

```ts
const response = await fetchOk(...);
return response.json();
```

Reply:
It also looks like the returned value isn't use anywhere, so alternatively, this function could be typed as returning a `Promise<void>` and only await fetchOk instead of returning it.
```
await fetchOk(...);
```